src/ripple/core/impl/Workers.cpp

```diff
@@ -144,27 +143,50 @@ void Workers::deleteWorkers (beast::LockFreeStack <Worker>& stack)
 //------------------------------------------------------------------------------
 
 Workers::Worker::Worker (Workers& workers, std::string const& threadName)
-    : Thread (threadName)
-    , m_workers (workers)
+    : m_workers {workers}
+    , threadName_ {threadName}
+    , wakeCount_ {0}
+    , shouldExit_ {false}
 {
-    startThread ();
+    thread_ = std::thread {&Workers::Worker::run, this};
 }
 
 Workers::Worker::~Worker ()
 {
-    stopThread ();
+    {
+        std::lock_guard <std::mutex> lock {mutex_};
+        ++wakeCount_;
+        shouldExit_ = true;
+    }
+
+    wakeup_.notify_one();
+    thread_.join();
+}
+
+void Workers::Worker::notify ()
+{
+    {
+        std::lock_guard <std::mutex> lock {mutex_};
+        ++wakeCount_;
+    }
+    wakeup_.notify_one();
```

Comment:
I recall David referencing an article a while back that claimed that doing the notify under the lock was actually a little better.  Though I don't think it is a big deal either way.  And I can't find a link to that article... :-\